In [4]:
import pandas as pd

pd.set_option('display.max_rows', 30)

### Loading the data dictionary from the file

In [5]:
# loading the data dictionary
file_name = 'institution_data_dictionary.csv'
df_data_dict_inst = pd.read_csv(file_name)
df_data_dict_inst.shape

(3251, 11)

### This shows that the data dictionary has 3251 rows. We can consider those as the features in our dataet.

In [35]:
# Displaying some sample values of the data dictionary
df_data_dict_inst

NAME OF DATA ELEMENT dev-category  \
0                               Unit ID for institution         root   
1                        8-digit OPE ID for institution         root   
2                        6-digit OPE ID for institution         root   
3                                      Institution name       school   
4                                                  City       school   
...                                                 ...          ...   
3246  Median earnings of students working and not en...     earnings   
3247  Median earnings of independent students workin...     earnings   
3248  Median earnings of dependent students working ...     earnings   
3249  Median earnings of non-male students working a...     earnings   
3250  Median earnings of male students working and n...     earnings   

                                developer-friendly name API data type  \
0                                                    id       integer   
1                                               ope8_id        string   
2                                               ope6_id        string   
3                                                  name  autocomplete   
4                                                  city  autocomplete   
...                                                 ...           ...   
3246  10_yrs_after_entry.median_earnings.highest_ter...       integer   
3247  10_yrs_after_entry.median_earnings.independent...       integer   
3248  10_yrs_after_entry.median_earnings.dependent_s...       integer   
3249  10_yrs_after_entry.median_earnings.non_male_st...       integer   
3250   10_yrs_after_entry.median_earnings.male_students       integer   

             INDEX           VARIABLE NAME VALUE LABEL    SOURCE  \
0              NaN                  UNITID   NaN   NaN     IPEDS   
1      varchar(10)                   OPEID   NaN   NaN     IPEDS   
2      varchar(10)                  OPEID6   NaN   NaN     IPEDS   
3         fulltext                  INSTNM   NaN   NaN     IPEDS   
4     varchar(200)                    CITY   NaN   NaN     IPEDS   
...            ...                     ...   ...   ...       ...   
3246           NaN    MD_EARN_WNE_INC3_P10   NaN   NaN  Treasury   
3247           NaN  MD_EARN_WNE_INDEP1_P10   NaN   NaN  Treasury   
3248           NaN  MD_EARN_WNE_INDEP0_P10   NaN   NaN  Treasury   
3249           NaN   MD_EARN_WNE_MALE0_P10   NaN   NaN  Treasury   
3250           NaN   MD_EARN_WNE_MALE1_P10   NaN   NaN  Treasury   

     SHOWN/USE ON SITE NOTES  
0                  Yes   NaN  
1                  Yes   NaN  
2                  Yes   NaN  
3                  Yes   NaN  
4                  Yes   NaN  
...                ...   ...  
3246               NaN   NaN  
3247               NaN   NaN  
3248               NaN   NaN  
3249               NaN   NaN  
3250               NaN   NaN  

[3251 rows x 11 columns]

In [37]:
### Getting to know which variables to pick for feature selection
df_data_dict_inst[['NAME OF DATA ELEMENT','VARIABLE NAME','VALUE','LABEL']][0:10]

NAME OF DATA ELEMENT VARIABLE NAME VALUE LABEL
0                     Unit ID for institution        UNITID   NaN   NaN
1              8-digit OPE ID for institution         OPEID   NaN   NaN
2              6-digit OPE ID for institution        OPEID6   NaN   NaN
3                            Institution name        INSTNM   NaN   NaN
4                                        City          CITY   NaN   NaN
5                              State postcode        STABBR   NaN   NaN
6                                    ZIP code           ZIP   NaN   NaN
7                  Accreditor for institution  ACCREDAGENCY   NaN   NaN
8              URL for institution's homepage       INSTURL   NaN   NaN
9  URL for institution's net price calculator        NPCURL   NaN   NaN

In [ ]:
# Load the most recent data file with the instituion names
data_file_inst = 'dataset/CollegeScorecard_Raw_Data_03142022/Most-Recent-Cohorts-Institution.csv'
df_inst = pd.read_csv(data_file_inst, low_memory=False)

In [90]:
df_inst.shape

(6694, 2989)

## Preparing the feature list from the dictionary

In [89]:
# Listing down all the features that I could think of initially in order to do my analysis

list_of_all_features=['INSTNM','CITY','STABBR','ST_FIPS','PREDDEG','ICLEVEL','CONTROL','ADM_RATE','CURROPER','SATVR25','SATVR75','SATMT25','SATMT75','SATWR25','SATWR75',
                 'ACTCM25','ACTCM75','ACTEN25','ACTEN75','ACTMT25','ACTMT75','ACTWR25','ACTWR75']

In [91]:
df_features_inst = df_inst[list_of_all_features]
df_features_inst.shape

(6694, 23)

In [72]:
df_features_inst_subset_1=df_features_inst.loc[[i for i in range(0,6694)],['INSTNM','CITY','STABBR','ST_FIPS','PREDDEG','CURROPER']]
df_features_inst_subset_1.shape

(6694, 6)

In [73]:
df_features_inst_subset_2=df_features_inst.loc[[i for i in range(0,6694)],['INSTNM','CURROPER']]
df_features_inst_subset_2.shape

(6694, 2)

In [74]:
byCurrOperation = df_features_inst_subset_2.groupby('CURROPER')
byCurrOperation.count()

INSTNM
CURROPER        
0            330
1           6364

### So this above output shows that there are 300 Institutions that are Not currently certified as an operating institution

In [92]:
# Filtering out the Institutions that are currently not operating
df_features_inst_curroper = df_features_inst.query("CURROPER == 1")
df_features_inst_curroper.shape

(6364, 23)

In [103]:
# Colleges by ADM rate
df_coll_adm_rate = df_features_inst_curroper.loc[0:,['INSTNM','CITY','PREDDEG','ICLEVEL','ADM_RATE']]
df_coll_adm_rate.shape

(6364, 5)

In [104]:
df_coll_adm_rate

INSTNM             CITY  PREDDEG  \
0                     Alabama A & M University           Normal        3   
1          University of Alabama at Birmingham       Birmingham        3   
2                           Amridge University       Montgomery        2   
3          University of Alabama in Huntsville       Huntsville        3   
4                     Alabama State University       Montgomery        3   
...                                        ...              ...      ...   
6689        Georgia Military College - Eastman          Eastman        0   
6690  American College of Barbering - Florence         Florence        0   
6691      HCI College - Fort Lauderdale Campus  Fort Lauderdale        0   
6692          ABC Adult School - Cabrillo Lane         Cerritos        0   
6693           Urban Barber College - San Jose         San Jose        0   

      ICLEVEL  ADM_RATE  
0           1    0.9175  
1           1    0.7366  
2           1       NaN  
3           1    0.8257  
4           1    0.9690  
...       ...       ...  
6689        1       NaN  
6690        3       NaN  
6691        2       NaN  
6692        3       NaN  
6693        3       NaN  

[6364 rows x 5 columns]

In [105]:
# Sort by ADM Rate
df_coll_adm_rate = df_coll_adm_rate.query("ADM_RATE > 0.0 & ADM_RATE < 1.0")
df_coll_adm_rate.sort_values(by='ADM_RATE', ascending=True, na_position='last')


INSTNM          CITY  PREDDEG  ICLEVEL  \
1464                   Hampshire College       Amherst        3        1   
2806           Curtis Institute of Music  Philadelphia        3        1   
3765                 Stanford University      Stanford        3        1   
1465                  Harvard University     Cambridge        3        1   
391                 Pacific Oaks College      Pasadena        3        1   
...                                  ...           ...      ...      ...   
2461          Dickinson State University     Dickinson        3        1   
842            Lewis-Clark State College      Lewiston        3        1   
800                    Morehouse College       Atlanta        3        1   
382       Notre Dame de Namur University       Belmont        3        1   
3354  The University of Texas at El Paso       El Paso        3        1   

      ADM_RATE  
1464    0.0197  
2806    0.0393  
3765    0.0434  
1465    0.0464  
391     0.0511  
...        ...  
2461    0.9973  
842     0.9976  
800     0.9980  
382     0.9994  
3354    0.9999  

[1854 rows x 5 columns]

In [98]:
df_coll_adm_rate = df_coll_adm_rate.query('INSTNM.str.contains("Indiana")', engine='python')
df_coll_adm_rate

INSTNM          CITY  \
1042  Indiana University-Purdue University-Indianapolis  Indianapolis   
1043                         University of Indianapolis  Indianapolis   
1044                    Indiana Institute of Technology    Fort Wayne   
1045                     University of Southern Indiana    Evansville   
1046                           Indiana State University   Terre Haute   
1047                          Indiana University-Kokomo        Kokomo   
1048                      Indiana University-South Bend    South Bend   
1049                     Indiana University-Bloomington   Bloomington   
1050                       Indiana University-Northwest          Gary   
1051                       Indiana University-Southeast    New Albany   
1052                            Indiana University-East      Richmond   
1054        International Business College-Indianapolis  Indianapolis   
1061                 Indiana Wesleyan University-Marion        Marion   
2846     Indiana University of Pennsylvania-Main Campus       Indiana   
5423                     Chamberlain University-Indiana  Indianapolis   
5905                           Indiana Wellness College  Indianapolis   

      PREDDEG  ICLEVEL  ADM_RATE  
1042        3        1    0.8073  
1043        3        1    0.9279  
1044        3        1    0.5964  
1045        3        1    0.9332  
1046        3        1    0.8957  
1047        3        1    0.7364  
1048        3        1    0.7770  
1049        3        1    0.7791  
1050        3        1    0.7462  
1051        3        1    0.8191  
1052        3        1    0.6252  
1054        2        2    0.7933  
1061        3        1    0.6700  
2846        3        1    0.9300  
5423        3        1    0.2727  
5905        1        3    0.8525

In [106]:
df_coll_adm_rate = df_coll_adm_rate.query('INSTNM.str.contains("bilt")', engine='python')
df_coll_adm_rate

INSTNM       CITY  PREDDEG  ICLEVEL  ADM_RATE
3200  Vanderbilt University  Nashville        3        1    0.0912